bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/gems/GemsVoTableCatalog.scala

```diff
@@ -88,23 +91,36 @@ case class GemsVoTableCatalog(backend: VoTableBackend = RemoteBackend) {
    * @param inst the instrument option for the search
    * @return a list of threads used for background catalog searches
    */
-  private def searchOptimized(basePosition: Coordinates, criterions: List[GemsCatalogSearchCriterion], inst: GemsInstrument, statusLogger: StatusLogger): Future[List[GemsCatalogSearchResults]] = {
+  private def searchOptimized(basePosition: Coordinates, conditions: Conditions, criterions: List[GemsCatalogSearchCriterion], inst: GemsInstrument, statusLogger: StatusLogger): Future[List[GemsCatalogSearchResults]] = {
     val radiusLimitsList = getRadiusLimits(inst, criterions)
     val magLimitsList = optimizeMagnitudeLimits(criterions)
 
     val queries = for {
       radiusLimits <- radiusLimitsList
-      magLimits <- magLimitsList
-      queryArgs = CatalogQuery.catalogQuery(basePosition, radiusLimits, magLimits.some)
+      magLimits    <- magLimitsList
+      mr            = MagnitudeRange(magLimits.faintnessConstraint, magLimits.saturationConstraint)
+      queryArgs     = (CatalogQuery.catalogQueryWithoutBand(basePosition, radiusLimits, mr.some), magLimits)
```

Comment:
This isn't "wrong" and I've done it many times before too so I'm not suggesting that you change anything here.  I'm wondering though in cases like this when the midstream definitions are not used until the `yield` if they should just be moved to the yield block.  I'm thinking the purpose of defining them midstream is to use them in subsequent filters and mappings, which isn't the case here.  Opinions?